ports/qemu-riscv/README.md

```diff
@@ -0,0 +1,38 @@
+This is experimental, community-supported port for RISC-V RV32IM emulation as
+provided by QEMU (http://qemu.org).
+
+The purposes of this port are to enable:
+
+1. Continuous integration
+    - run tests against architecture-specific parts of code base
+2. Experimentation
+    - simulation & prototyping of anything that has architecture-specific
+      code
+    - exploring instruction set in terms of optimising some part of
+      MicroPython or a module
+3. Streamlined debugging
+    - no need for JTAG or even an MCU chip itself
+    - no need to use OpenOCD or anything else that might slow down the
+      process in terms of plugging things together, pressing buttons, etc.
+
+This port requires a newlib based bare metal/ELF RISC-V toolchain, either
+with multilib support or 32 bits specific (M, C, and Zicsr extensions must
+be supported, along with ilp32 ABI).  Either musl-based or glibc-based
+toolchains are not tested or known to work.
+
+If your toolchain does not support the above requirements, either download
+a pre-built toolchain (and that's a hit or miss situation in its own right)
+or follow the instructions on [RISC-V's reference toolchain Github repo](https://github.com/riscv-collab/riscv-gnu-toolchain)
+to get a newlib-based multilib toolchain (the target will be
+`riscv64-unknown-elf` but said toolchain will be able to generate 32 bits
+code as well).  If you plan to use the reference toolchain, use the
+[2023.10.06](https://github.com/riscv-collab/riscv-gnu-toolchain/tree/2023.10.06)
+tag, based on GCC version 12.2.0.  Later versions, based on GCC 13.2.0 generate
+code that makes test fail in certain conditions.
```

Comment:
Maybe you can add the commands from your comment in the PR to build the toolchain?

Reply:
Good point, I'll add that.